optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,474 @@
+            cache_dir (`Union[str, Path]`, *optional*):
+                The path to a directory in which a downloaded pretrained model configuration should be cached if the
+                standard cache should not be used.
+            force_download (`bool`, defaults to `False`):
+                Whether or not to force the (re-)download of the model weights and configuration files, overriding the
+                cached versions if they exist.
+            file_name(`str`, *optional*):
+                The file name of the model to load. Overwrites the default file name and allows one to load the model
+                with a different name.
+            local_files_only(`bool`, *optional*, defaults to `False`):
+                Whether or not to only look at local files (i.e., do not try to download the model).
+        """
+        backend = getattr(config, "backend", None)
+
+        if getattr(config, "torchscript", None) or backend == "neural_engine":
+            # Load the model from local directory
+            if os.path.isdir(model_id):
+                if backend == "neural_engine":
+                    file_name = model_id
+                else:
+                    file_name = os.path.join(model_id, file_name)
+                    model_save_dir = model_id
+            # Download the model from the hub
+            else:
+                if backend == "neural_engine":
+                    model_file_names = {"model": ENGINE_MODEL_NAME, "config": ENGINE_MODEL_CONFIG}
+                    try:
+                        for name, file_name in model_file_names.items():
+                            model_cache_path = hf_hub_download(
+                                repo_id=model_id,
+                                filename=file_name,
+                                use_auth_token=use_auth_token,
+                                revision=revision,
+                                cache_dir=cache_dir,
+                                force_download=force_download,
+                                local_files_only=local_files_only,
+                            )
+                        file_name = Path(model_cache_path).parent
+                    except Exception:
+                        logger.warning(
+                            f"The file names {ENGINE_MODEL_NAME} or {ENGINE_MODEL_CONFIG} was not found! Please check it!"
+                        )
+                        raise Exception
+                else:
+                    model_cache_path = hf_hub_download(
+                        repo_id=model_id,
+                        filename=file_name,
+                        use_auth_token=use_auth_token,
+                        revision=revision,
+                        cache_dir=cache_dir,
+                        force_download=force_download,
+                        local_files_only=local_files_only,
+                    )
+                model_save_dir = Path(model_cache_path).parent
```

Comment:
This could be refactor by calling `hf_hub_download` once (only updating `model_file_names`)